ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analysis.java

```diff
@@ -437,6 +437,14 @@ public Expression getRightJoinExpression() {
       return rightJoinExpression;
     }
 
+    public Expression getFlippedLeftJoinExpression() {
+      return flippedJoinCondition ? rightJoinExpression : leftJoinExpression;
```

Comment:
I did not review the previous PR adding this, but regarding the flipped flag, if we stick with the current semantics it seems we would not allow such flipping in FK-join since we need to stick to which PK to inherit from the join (see https://github.com/confluentinc/ksql/pull/7537#discussion_r635482988 point 0), please correct me if my understanding is wrong?

Reply:
@guozhangwang there is a distinction between flipping the left and right sources for the join (i.e., the difference between LEFT JOIN and RIGHT JOIN) versus flipping the order of the join expressions (i.e., the difference between `JOIN ON a=b` and `JOIN ON b=a`, without flipping the order of the join sources). ksqlDB does not allow the former but _does_ allow the latter.

Reply:
Got it, I think I was originally misunderstanding this, because the flipping's purpose is to make sure the new source of a multi-way join is always a source. Now I understand the flipping is only on the `JOIN`.